### DISCUSSION TITLE
Extend attack time

### DISCUSSION THREAD

**Discussion by Stepzor11** [risinek/esp32-wifi-penetration-tool#56] [Q&A]
As far as I can see, the maximum attack time is 255 seconds.
It would be nice to extend it further.
Can you please help me?

**Comment by risinek** [comment]
The timeout is currently limited by the unit8 type https://github.com/risinek/esp32-wifi-penetration-tool/blob/e346e28ed9c93e98ab686c30eb8c4bc549313ba3/main/attack.h#L49
if you want to use larger value, you will have to use larger data type.  I would suggest unit16 or unit32.

You will then need to adjust the parsing functions on the webserver, because currently it expects single byte.
https://github.com/risinek/esp32-wifi-penetration-tool/blob/e346e28ed9c93e98ab686c30eb8c4bc549313ba3/components/webserver/utils/index.html#L236
and change the limit in HTML https://github.com/risinek/esp32-wifi-penetration-tool/blob/e346e28ed9c93e98ab686c30eb8c4bc549313ba3/components/webserver/utils/index.html#L57

Feel free to open PR when you have something and I can try to help you if you are stuck. I think prolonging timeout would be welcomed improvement by others.

**Reply by Stepzor11** [reply]
Amazing, thanks!

I am not enough skilled in programing.
I'll try to ask to some friends hoping that they are expert enough to do that.

Of course, if I gave you a good idea and you can consider to to improve it by yourself, it would be awesome, because you know the code better than all of us.

Just let me know if I triggered your intetrest enough to improve the project directly by yourdìself, so I'll stop asking friends for helping me

**Reply by risinek** [reply]
Sorry, I don't have time for this project anymore. At least not at the moment. If you can manage to find someone, contribution is always welcome.
This project was mostly about proving that it is possible to run attacks on Wi-Fi networks with ESP32. So improvements like prolonged timeouts have minor priority as it does not add any value into research.

**Reply by azhar25785** [reply]
i made these changes now how i add these changes to build files

**Reply by Stepzor11** [reply]
> i made these changes now how i add these changes to build files

Have you found a solution?

**Comment by Stepzor11** [comment]
No problem, we will try our best

**Reply by risinek** [reply]
Good luck 👍

**Reply by azhar25785** [reply]
hey  man i need at least 30mins time .... can you please upgrade the code it'll be a great help

**Reply by vanke88** [reply]
If the author has time, the following features can be updated:
1.If the attack duration is set to 0, the attack will not stop until it is manually disabled, and multiple APs can be selected to take turns attacking
2. The download button will not be displayed if the handshake package capture fails, 
3. The download button will be displayed immediately after the handshake package capture is successful instead of waiting for the timer to end.
4. Whether the handshake package is captured successfully or not, there is a corresponding text prompt, instead of showing white or a string of characters. These features have been updated and sold on hardware, but I can't find the code
![8962b272c7383efd218fe0d7ffb6c4b](https://github.com/risinek/esp32-wifi-penetration-tool/assets/141999670/d3071ac1-814e-494f-b080-826bcf7ec7ae)

**Comment by vanke88** [comment]
[index.zip](https://github.com/risinek/esp32-wifi-penetration-tool/files/14754622/index.zip)
This is the homepage code I got, and I can't get the other codes
